addons/website_slides_survey/__init__.py

```diff
@@ -3,3 +3,11 @@
 
 from . import models
 from . import controllers
+
+from odoo import api, SUPERUSER_ID
+
+def uninstall_hook(cr, registry):
+    env = api.Environment(cr, SUPERUSER_ID, {})
+    dt = env.ref('website_slides.badge_data_certification_goal')
```

Comment:
```suggestion
    dt = env.ref('website_slides.badge_data_certification_goal', raise_if_not_found=False)
```
And the check if dt actually exist